### DISCUSSION TITLE
Camera changes aspect ratio

### DISCUSSION THREAD

**Discussion by thesebastianf** [blakeblackshear/frigate#14446] [Ask A Question] [answered]
Hey I couldnt find what is the reason why my cameras:
- show by chance a aspect ratio with no borders
- sometimes with black borders left right
- sometimes "jump" from with borders to no borders a few seconds later

any hint what setting I can look at would be much appreciated

![frigate_borders](https://github.com/user-attachments/assets/ffefdd27-5407-4765-b454-7fd269b6f361)

**Comment by hawkeye217** [comment]
Without seeing your config it's difficult to tell you exactly, but usually this is usually a result of the stream that you're using for the `detect` role being a different aspect ratio than the one you're using for live view.

**Reply by thesebastianf** [reply]
you are right
thats when no motion is detected and a still image is shown ... which is then seemingly from the substream (detect)
My cameras dont allow the same resolution for the substream which I use for detect
Can I enforce that the still image is from the main stream or scale it to another resolution?

**Reply by NickM-27** [reply, marked as answer]
You can change the detect width / height to a different aspect ratio to adjust it

**Reply by darnmason** [reply]
Ooh this is so much better, it has stretched the image back to "normal". I had been putting up with the aspect ratio mismatch.

I'm unclear if there are any repercussions now it doesn't match the native stream resolution?

For reference my record stream is 3072x2048 so 1.5 whereas my substream maxes out at 640x480 so 1.33
I've change it to 720x480 to maintain the aspect ratio.

**Comment by jsapede** [comment]
got the same problem on 0.15.

aspect ratio in detect is set to 1080p, camera flux is 1920x1080 : 

```
##################
# Options Detect #
##################
detect:
  enabled: true
  width: 1920
  height: 1080
  fps: 5
  stationary:
    interval: 50
    threshold: 50
```

aspect ratio is ok on recordings : 

![image](https://github.com/user-attachments/assets/f7c6649b-5b1f-449a-9acb-785abc85fbac)

but NOK on live : 

![image](https://github.com/user-attachments/assets/4c282f45-d4fe-4599-9133-998cafbe5b70)

trie to change the settings il nive section 

```
# Reglages live
live:
  height: 1080
  quality: 20
```

but without any effects. tried to change in detection but also with no effect

had no problems with previous versions of frigate

**Comment by jsapede** [comment]
digging al little i found that te problme comes from MSE stream.

Looking a stream in go2rtc, form pc at work, stream is MSE : 

![image](https://github.com/user-attachments/assets/c7bf65b3-9acb-4792-ae4b-c742c969738d)

impossible to have webrtc working at work. So looking on the phone, with webrtc access got the good aspect ratio : 

![image](https://github.com/user-attachments/assets/5c108315-4c78-4872-929e-ea027d6ae021)

**Reply by hawkeye217** [reply]
Since you can reproduce this with the streams on your go2rtc dashboard, this rules out any issue with Frigate's code. 

First thing I would ensure would be the codec being sent from your camera is H.264.

**Reply by jsapede** [reply]
yes sure, as it seemes to come from go2rtc !
'ill investigate but got the same phenomenon on 2 types of camera, and its was working normally some weeks ago.

**Reply by hawkeye217** [reply]
If it was working before, perhaps it is caused by a browser update? You could try transcoding and specifying the width and height:

```
go2rtc:
  streams:
    camera:
      - ffmpeg:rtsp://...#video=h264#width=1920#height=1080#audio=copy#hardware
```

**Reply by jsapede** [reply]
this is relevant ! didint think about his ! 
tested from Phyrox and aspect ratio is good :

![image](https://github.com/user-attachments/assets/6836178f-76e2-4b49-bb28-a9cee90145f0)

thanks !

**Reply by hawkeye217** [reply]
Great, happy to help.

**Reply by hawkeye217** [reply]
Several other users have noted aspect ratio issues with Chrome in particular. I'd be curious if you tried Firefox or another non-Chromium based browser to see if this improves without the go2rtc transcoding.

**Reply by jsapede** [reply]
phyrox is a portable fork of firefox, and screens were taken wtihout transcoding

**Reply by tobi-github** [reply]
Transcoding could be problematic on weaker CPUs. It should actually be sufficient to simply copy the video and audio stream through ffmpeg, then the aspect ratio should be correct again:
`ffmpeg:rtsp://...#video=copy#audio=copy`

The problem should affect Chrome-based browsers (according to @jsapede, this only affects the MSE stream).

From the Frigate user's point of view, however, this is unfortunately a non-deterministic behaviour, as neither the version of Frigate nor the configuration have been changed and therefore this behaviour suddenly appears. 
_Of course, this behaviour is not really non-deterministic, because it results from an update of the browser (engine) - which in most cases happens without conscious intervention by the user._ 

You may want to document this behaviour and possibly push the streams internally through ffmpeg as a hotfix (copy/[copy])

**Reply by cplemaster** [reply]
> Transcoding could be problematic on weaker CPUs. It should actually be sufficient to simply copy the video and audio stream through ffmpeg, then the aspect ratio should be correct again: `ffmpeg:rtsp://...#video=copy#audio=copy`
> 
> The problem should affect Chrome-based browsers (according to @jsapede, this only affects the MSE stream).
> 
> From the Frigate user's point of view, however, this is unfortunately a non-deterministic behaviour, as neither the version of Frigate nor the configuration have been changed and therefore this behaviour suddenly appears. _Of course, this behaviour is not really non-deterministic, because it results from an update of the browser (engine) - which in most cases happens without conscious intervention by the user._
> 
> You may want to document this behaviour and possibly push the streams internally through ffmpeg as a hotfix (copy/[copy])

Using this band-aid fixed it for me. Thanks so much! It was only one camera doing this so it must be something about the way OpenIPC is encoding the substream. Not sure. Was driving me crazy, thanks a lot.
